### DISCUSSION TITLE
How to use DDP + FP16 without fit?

### DISCUSSION THREAD

**Discussion by turian** [Lightning-AI/pytorch-lightning#6468] [DDP / multi-GPU / multi-node]
We have a model that doesn't require fit(). We just want to run inference with it using DDP + FP16.

Shoe-horning it into lightning is a little strange, but it gives use DDP + FP16 very easily. However, it appears that we need a dummy fit() (with dummy optimizer and train_step).

Without fit(), we get strange errors just invoking test. How can we use DDP + FP16 without fit?

**Comment by turian** [comment]
Seems to work?

**Reply by carmocca** [reply]
I'm not sure I get what you want

If what you want is to run `test`, you do not need to run `fit` before

```python
model  = ...
trainer = Trainer(accelerator="ddp", precision=16)
trainer.test(model)
```

Should work with no issues
